### PR TITLE
feature(plugins/core): Generic Results

### PR DESCRIPTION
**Pull request by soyacz**
Introduced generic results: allows clients to send arbitrary tabular
data to Argus and show it in `Results` tab in run view.
See [docs/generic_results.md](./docs/generic_results.md) for more details.

refs: https://github.com/scylladb/qa-tasks/issues/1244

### CODE REVIEW

**Comment by soyacz** [comment]
Example result:
image

File: frontend/TestRun/ResultsTab.svelte [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            console.log(error);
+        }
+    };
+
+    const durationToStr = (value) => {
+        const hours = Math.floor(value / 3600);
+        const minutes = Math.floor((value % 3600) / 60);
+        const seconds = value % 60;
+        return `${hours.toString().padStart(2, "0")}:${minutes.toString().padStart(2, "0")}:${seconds.toString().padStart(2, '0')}`;
+    };
```
Comment (k0machi) [inline frontend/TestRun/ResultsTab.svelte:31]: We have [humanize-duration](https://www.npmjs.com/package/humanize-duration) as a dependency, you can use that instead.

File: argus/backend/service/testrun.py [hunk 2]
```diff
@@ -297,6 +298,25 @@ def get_run_events(self, run_id: UUID):
         }
         return response
 
+    def fetch_results(self, test_id: UUID, run_id: UUID) -> dict:
+        query_fields = ["column", "row", "value", "status"]
+        tables_meta = ArgusTestResultMetadata.filter(test_id=test_id)
+        tables = []
+        print(tables_meta)
```
Comment (k0machi) [inline argus/backend/service/testrun.py:305]: Reminder

**Review by k0machi — COMMENTED** [review]
Looks good so far, just one comment.

**Comment by soyacz** [comment]
Ok, I made final fixes and I think it's ready for final review.

File: docs/generic_results.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+result_table.add_result(column="latency", row="mean", value=1.1, status=Status.WARNING)
+result_table.add_result(column="op_rate", row="mean", value=59988, status=Status.ERROR)
+result_table.add_result(column="latency", row="p99", value=2.7, status=Status.PASS)
+result_table.add_result(column="op_rate", row="p99", value=59988, status=Status.WARNING)
+result_table.add_result(column="duration", row="p99", value=8888, status=Status.ERROR)
+result_table.add_result(column="duration", row="mean", value=332, status=Status.PASS)
+
+# when testing locally, otherwise just use ArgusClient instance in your test.
+run_id = UUID("24e09748-bba4-47fd-a615-bf7ea2c425eb")
+client = ArgusSCTClient(run_id, auth_token="UO+2GXL9XqSgcVJijWk5WnbPXPit5ot5nfkLAHAr7SaqROfSCWycabpp/wxyY8+I",
```
Comment (fruch) [inline docs/generic_results.md:51]: is this a real token ?

Comment (soyacz) [reply docs/generic_results.md:51]: no, just my local development one. I revoked it and remove from here too.

Comment (fruch) [reply docs/generic_results.md]: Good

**Comment by soyacz** [comment]
still, few bugs. Updated table look slighlty:
image

File: argus/client/generic_result.py
Comment (soyacz) [inline argus/client/generic_result.py:1]: this will require new client - @k0machi how to prepare that?

**Comment by fruch** [comment]
> still, few bugs. Updated table look slighlty: image

this looks good, but after you left the sync meeting today
we discussed the need of being able to copy this table out, as markdown (for github issues)

**Comment by soyacz** [comment]
> this looks good, but after you left the sync meeting today we discussed the need of being able to copy this table out, as markdown (for github issues)

When selecting whole table (and maybe just a bit more) then I'm capable of putting it to gh (need to remove first line with some styles).
Also I was able to paste it to google spreadsheet (with CTRL+SHIFT+V).
I could also think of 'copy' button next to each table.

File: argus/client/generic_result.py
Comment (fruch) [reply argus/client/generic_result.py:1]: I know it's get uploaded to pypi...

but how, and whom has permissions.. 

so @k0machi , first all us to the pypi project, and take the bus factor down.

this is my username to add:
- https://pypi.org/user/fruch-scylladb/

2nd, enabling 2FA for all of pypi account in that project

3rd, write the setup of upload to pypi (do we tag master when we do that ?)

4th, we should automate push to pypi with github actions when we tag

@soyacz for development, you can put your branch/fork directly in SCT requirements.in, and build and image with it.
we can think of a shorter way, and install it during hydra executing of command

we have similar option in dtest for installing any driver version (including fork/branch)

**Comment by fruch** [comment]
> > this looks good, but after you left the sync meeting today we discussed the need of being able to copy this table out, as markdown (for github issues)
> 
> When selecting whole table (and maybe just a bit more) then I'm capable of putting it to gh (need to remove first line with some styles). Also I was able to paste it to google spreadsheet (with CTRL+SHIFT+V). I could also think of 'copy' button next to each table.

yes, the whole talk started from the copy button we have for the issue lists

File: argus/backend/util/config.py [hunk 4]
```diff
@@ -12,6 +12,7 @@ class Config:
     CONFIG_PATHS = [
         Path("./config/argus_web.yaml"),
         Path("argus_web.yaml"),
+        Path("../argus_web.yaml"),
```
Comment (k0machi) [inline argus/backend/util/config.py:15]: Do we need this for this PR or is this something QoL related?

File: argus/client/generic_result.py
Comment (k0machi) [reply argus/client/generic_result.py:1]: Will do, don't forget to edit `pyproject.toml` for your development version (use `b1`...`bN` designation for those)

File: frontend/TestRun/ResultsTab.svelte [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    table {
+        width: auto;
+        table-layout: auto;
+    }
+</style>
+
+{#if !fetching}
+    <div class="m-2">
```
Comment (k0machi) [inline frontend/TestRun/ResultsTab.svelte:74]: Looking at the screenshot, you may want to change this to `mb-2` or no margin at all, there's a grey line under the tab marker on yours.

File: frontend/TestRun/ResultsTab.svelte [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        {#each results as result}
+            <div class="">
+                <h3>{result.meta.name}</h3>
+                <span>{result.meta.description}</span>
+                <table class="table table-bordered">
+                    <thead class="thead-dark">
+                    <tr>
+                        <th></th>
+                        {#each result.meta.columns_meta as col}
+                            <th>{col.name} [{col.unit}]</th>
```
Comment (k0machi) [inline frontend/TestRun/ResultsTab.svelte:84]: can try something like this instead here
```html
<span class="rounded-pill d-inline-block bg-secondary" style="font-weight: 0.8em">{col.unit}</span>
```

File: frontend/TestRun/TestRun.svelte [hunk 7]
```diff
@@ -275,6 +278,16 @@
                         role="tab"
                         ><Fa icon={faExclamationTriangle}/> Activity</button
                     >
+                    <button
```
Comment (k0machi) [inline frontend/TestRun/TestRun.svelte:281]: Maybe put it somewhere in the middle, for example next to the potential jUnit tab.

**Review by k0machi — CHANGES_REQUESTED** [review]

File: argus/backend/util/config.py (hunk 4 above)
Comment (soyacz) [reply argus/backend/util/config.py:15]: QoL only

File: argus/client/generic_result.py
Comment (k0machi) [reply argus/client/generic_result.py:1]: Next available version will be `0.12.5`, `.4` is taken by driver matrix rework

**Comment by soyacz** [comment]
@k0machi can we deploy this PR to staging so I could test it with SCT (and implement couple generic-results sending to Argus in the wild)?
Also I'd test client properly then.

File: frontend/TestRun/TestRun.svelte (hunk 7 above)
Comment (soyacz) [reply frontend/TestRun/TestRun.svelte]: moved

File: frontend/TestRun/ResultsTab.svelte (hunk 6 above)
Comment (soyacz) [reply frontend/TestRun/ResultsTab.svelte]: I didn't like it. But emphasized a bit - see next screenshot.

**Comment by soyacz** [comment]
After minor fixes:
image

File: frontend/TestRun/ResultsTab.svelte (hunk 5 above)
Comment (soyacz) [reply frontend/TestRun/ResultsTab.svelte]: i needed also to be moved a bit from the left edge. Added.

File: argus/backend/util/config.py (hunk 4 above)
Comment (soyacz) [reply argus/backend/util/config.py]: removed

File: docs/generic_results.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+If using different SUT than ScyllaDB (or need to adjust dynamically) you can set `sut_timestamp` field in table constructor. `sut_timestamp` should always correspond to SUT build timestamp. Like this:
+```python
+result_table = LatencyResultTable(sut_timestamp=1629302400)
+```
+This will assure proper sorting of results in Argus UI when comparing to different versions in time based graphs.
+
+# Limitations
+* Only 3 types of values are supported: FLOAT, INTEGER, DURATION
+* automatic SUT timestamp supported only by SCT: it calculates it from Scylla Version date and is not correct for 2 different builds from the same date.
```
[the remark this answers is not in the document]
Comment (soyacz) [reply docs/generic_results.md]: I made it more clear.

**Review by k0machi — APPROVED** [review]
